src/API.jl

```diff
@@ -917,9 +925,21 @@ function precompile(ctx::Context)
             break
         end
         if stale
-            Base.compilecache(pkg, sourcepath)
+            t = @async begin
+                if length(pkg_dep_lists[i]) > 0 # if pkg has deps
+                    while length(precomp_list) == 0 || !all(map(x->in(x, precomp_list), pkg_dep_lists[i])) # if all deps not precomped
+                        sleep(0.001)
+                    end
```

Comment:
Spinning in loops like this is probably adding precious milliseconds to your precompile time.  Yes, it's a tiny fraction of the overall time, but you could use `Event()`'s to signify that a dep is done being precompiled.  You build a `Dict{PkgId,Event}()`, then in each `Task`, iterate over all deps and `wait()` upon their deps.  Once you're done precompiling your module, you `notify()` your event and you're done!  No more spinning.

Reply:
Oh nice. Yeah that's better